ClangSharpPInvokeGenerator/Program.cs

```diff
@@ -83,13 +83,18 @@ public static int Run(InvocationContext context)
                 return -1;
             }
 
-            string[] arr = { "-x", "c++" };
+            var arr = new string[]
+            {
+                "-xc++",                                // The input files are C++
+                "-Wno-pragma-once-outside-header"       // We are processing files which may be header files
```

Comment:
This is the one warning I think is common enough to always suppress.

Windows header files in general (e.g those from the Windows SDK) have a few other common warnings that it might be nice to provide a switch for easy suppression (such as #include directives using non matching casing).